src/arrayMethodSort.js

```diff
@@ -4,8 +4,23 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(
+    compareFunction = (a, b) => String(a) > String(b)
+  ) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - i - 1; j++) {
+        const compareResult = compareFunction(this[j], this[j + 1]);
+
+        if (compareResult > 0) {
```

Comment:
The comparison inside the 'if' statement is not correct. The compareFunction should return a negative, zero, or positive value, dependent on the arguments. You should check if compareResult is more than 0.

Reply:
https://mate.academy/chats/27190?selectedMessageId=1401459